sklearn/feature_extraction/_hashing.pyx

```diff
@@ -45,7 +43,7 @@ def transform(raw_X, Py_ssize_t n_features, dtype):
 
     for x in raw_X:
         for f, v in x:
-            if isinstance(v, string_types):
+            if isinstance(v, basestring):
```

Comment:
This requires Cython 0.20+, is it OK? If not, it is possible to replace it with (str, unicode) which does the same.